magento/magento2
# #14020-Cart-Sales-Rule-with-negated-condition-over-special-price-does…

…-not-work-for-configurable-products.

Use configurable product`s children for shopping cart rules validation
for cases when attribute exists for children only (E.g. special_price)





### Description


### Fixed Issues (if relevant)

1. magento/magento2#14020 Cart Sales Rule with negated condition over special_price does not work for configurable products

### Manual testing scenarios

1. Enable the `special_price` for promo rule conditions: 
   1. Stores > Attributes > Product
   2. Search for and edit `special_price`
   3. Storefront Properties: Use for Promo Rule Conditions: Yes
   4. Save
2. create new cart sales rule:
   1. Marketing > Cart Price Rules
   2.  Add New Rule:
       1. Rule name = "Special Price Test"
       2. Website: select all
       3. Customer Groups: select all
       4. Coupon: "Specific Coupon"
       5. Coupon Code: "special"
       6. Actions: Discount Amount: "10"
       7. Actions: Apply the rule only to cart items matching the following conditions: "if **ALL** of these conditions are **FALSE**": "Special Price **greater than 0.01**"
       8. Leave all other fields to their defaults and save
3. In the front end, add a the child product of a configurable product to the cart, such that the child product has a special price set. This product should be the only one in the cart.
4. "View and Edit Cart" > "Apply Discount Code": `special`

### Contribution checklist
